### ISSUE TITLE
Done Script Not Running

### ISSUE THREAD

**Issue by maxtimbo** [transmission/transmission#1555]
I know this issue has been opened before, but I can't seem to find any definitive answers. I have Transmission running inside a jail on TrueNAS 12. I have enabled 

```json
"script-torrent-done-enabled": true,
"script-torrent-done-filename": "/usr/local/etc/transmission/home/start_script.sh",
```

start_script.sh:

```sh
#!/bin/sh
python3 /usr/local/etc/transmission/home/mv_completed.py
```

Currently mv_completed.py:

```python
#!/usr/bin/env python3

import os, shutil, urllib.request

downloads_src = "/media"
downloads_dst = "/mnt/completed"
log_dir = "log"

app_vers = os.environ.get("TR_APP_VERSION")
local_time = os.environ.get("TR_TIME_LOCALTIME")
tor_dir = os.environ.get("TR_TORRENT_DIR")
tor_hash = os.environ.get("TR_TORRENT_HASH")
tor_id = os.environ.get("TR_TORRENT_ID")
tor_name = os.environ.get("TR_TORRENT_NAME")

ext_ip = urllib.request.urlopen('http://ident.me').read().decode('utf8')

log = f"transmission{local_time}.log"

log_path = os.path.join(downloads_dst, log_dir, log)

with open(log_path, 'a+') as f:
    f.write(f'{local_time}: {ext_ip}\n')
    f.write(f'{local_time}: {app_vers}\n')
    f.write(f'{local_time}: {tor_dir}\n')
    f.write(f'{local_time}: {tor_hash}\n')
    f.write(f'{local_time}: {tor_id}\n')
    f.write(f'{local_time}: {tor_name}\n')
    f.write('--------------\n')
    f.close()
```

As you can see, all I want right now is to write the environment variables sent from transmission to a log file. Nothing happens when a torrent has completed. 

So my question is A) How can I test an 'on-complete' script without an actual torrent (if possible). B) what am I doing wrong, if anything? what's missing?

**Comment by mikedld** [comment]
A basic checklist of what [not] to expect when writing completion scripts could be found in https://github.com/transmission/transmission/issues/405#issuecomment-472212932. It was written for Windows but most points aren't OS-specific.

For your particular case, don't use relative paths (or set current working directory explicitly beforehand), and check that transmission user (that your daemon is running as, and that your script is being executed as) has access to where it tries to write the log. To catch early Python errors you may want to redirect to log in start_script.sh instead, e.g.

```sh
#!/bin/sh
python3 /usr/local/etc/transmission/home/mv_completed.py > /tmp/mv_completed.log 2>&1
```

**Comment by mikedld** [comment]
Please use [forums](https://forum.transmissionbt.com/) for support.

**Comment by maxtimbo** [comment]
> Please use [forums](https://forum.transmissionbt.com/) for support.

The forums are often broken. This click was the first time I've successfully opened them, but it was an offline version. So kinda useless.

Anyways, the problem was permissions. TrueNAS creates a user in the jail that has no privileges, shell is /bin/nologin (or something like that, I forget exactly). 

What I did to solve my particular issue is to just spin up a Debian VM and run Transmission on that. Works much better than the jail anyways.
